### PR TITLE
Hardening nginx, allow TLSv1.2 with the most secure cipher suites only

### PR DESCRIPTION
**Pull request by c4539**

### CODE REVIEW

**Comment by kspearrin** [comment]
Any possible compatibility issues from this change?

**Comment by c4539** [comment]
Yes, older clients will no longer be able to connect:

handshake_simulation
protocol_mismatch

**Comment by c4539** [comment]
However, I think it's reasonable to use the best cipher suits available, because Bitwarden is dealing with the most sensitive data.

File: util/Setup/NginxConfigBuilder.cs [hunk 1]
```diff
@@ -7,11 +7,8 @@ public class NginxConfigBuilder
     {
         private const string ConfFile = "/bitwarden/nginx/default.conf";
         private const string SslCiphers =
-            "ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:" +
-            "DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA:" +
-            "ECDHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:" +
-            "ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES256-GCM-SHA384:AES128-GCM-SHA256:AES256-SHA256:" +
-            "AES128-SHA256:AES256-SHA:AES128-SHA:DES-CBC3-SHA:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4:@STRENGTH";
+            "ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:" +
```
Comment (kspearrin) [inline util/Setup/NginxConfigBuilder.cs]: I would prefer that we follow the guidance from Mozilla here: https://wiki.mozilla.org/Security/Server_Side_TLS

Also, we must support Android 4.4, so I am not sure which ciphers are required there.

Comment (kspearrin) [reply util/Setup/NginxConfigBuilder.cs]: Looks like they have a neat generator tool here:
https://mozilla.github.io/server-side-tls/ssl-config-generator/

**Comment by c4539** [comment]
I added some more ciphers, so any well known browsers that supports TLS1.2 can establish a connection.
I still wouldn't recommend to keep TLS1 and 1.1 enabled

handshake_simulation2

**Comment by kspearrin** [comment]
Could you share what tool you are using in those screenshots?

**Comment by kspearrin** [comment]
Thanks.

**Comment by Jacalz** [comment]
How about supporting TLS 1.3?

**Comment by c4539** [comment]
Yes, TLSv1.3 should be supported as well. I haven't included it in my change because the latest nginx docker image is built with OpenSSL 1.1.0f, which does not support TLSv1.3. So it requires a bit more work.

**Comment by Jacalz** [comment]
Okay, thanks for the information 👍

**Pull request merged**